### ISSUE TITLE
sks-keyserver: dump keys to mirrors

### ISSUE THREAD

**Issue by abizer** [ocf/puppet#474]
there aren't that many high-quality sks keyserver keydumps available for starting new servers from. most of the existing ones are low-bandwidth, slow, or unreliable at best, so I think it would be valuable for us to offer a keydump. We can put it on mirrors and automatically get http/s, rsync, and ftp as well, at a cost of just ~12GB of disk space.

Here is my tentative proposal for how to do it, but I'm not sure it's a good or tractable design.

some constraints:
1. The keyserver itself needs to be turned off during the dump (~3 minutes)
2. The dump command (`sks dump $count $output_directory $output_prefix`) is a little too smart - it tries to split the dump into separate files (each with $count keys) and write them to the specified output directory itself, rather than dumping to a single file / stdout and letting us split/compress it ourselves afterwards.

my proposal would be something like the following:

write a `sync-archive`-style script in `ocf_mirrors` that uses a keytab like `ocfbackups` to log into the keyserver and:
1. `rm -r /var/lib/sks/dump/*`
2. `systemctl stop sks-keyserver`
3. `docker run --rm -it -w /var/lib/sks -v /var/lib/sks:/var/lib/sks zhusj/sks sks dump 20000 /var/lib/sks/dump sks-dump`
4. `systemctl start sks-keyserver`
then, back on the host, `rsync -azH --delete something@pgp:/var/lib/sks/dump /opt/mirrors/ftp/sks-keydump/`

what do y'all think?

**Comment by dkess** [comment]
Why does the keyserver need to be off during this? Can't we make it read-only?

**Comment by abizer** [comment]
nope, here's the wiki on the issue: https://bitbucket.org/skskeyserver/sks-keyserver/wiki/DumpingKeys

I've tried reading the source but I've had minimal luck decoding it and figuring out why it's so messed up
